libsession/src/main/java/org/session/libsession/messaging/jobs/AttachmentDownloadJob.kt

```diff
@@ -40,6 +42,31 @@ class AttachmentDownloadJob(val attachmentID: Long, val databaseMessageID: Long)
         // Keys used for database storage
         private val ATTACHMENT_ID_KEY = "attachment_id"
         private val TS_INCOMING_MESSAGE_ID_KEY = "tsIncoming_message_id"
+
+        fun eligibleForDownload(threadID: Long,
+                                storage: StorageProtocol,
+                                messageDataProvider: MessageDataProvider,
+                                databaseMessageID: Long): Boolean {
+            val threadRecipient = storage.getRecipientForThread(threadID)
+            val selfSend = messageDataProvider.isMmsOutgoing(databaseMessageID)
+            val sender = if (selfSend) {
+                storage.getUserPublicKey()
+            } else {
+                messageDataProvider.getIndividualRecipientForMms(databaseMessageID)?.address?.serialize()
+            }
+
+            val contact = sender?.let { storage.getContactWithSessionID(it) }
+            if (threadRecipient == null || sender == null || (contact == null && !selfSend)) {
+                return false
+            }
+            if (!threadRecipient.isGroupRecipient && contact?.isTrusted != true && storage.getUserPublicKey() != sender) {
```

Comment:
Would it simpler, like this?

```suggestion
            if (!threadRecipient.isGroupRecipient && contact?.isTrusted != true && !selfSend) {
```

or even:

```suggestion
            return threadRecipient.isGroupRecipient || contact?.isTrusted == true || selfSend
```

actually, that is a big mental unlock for me understanding this easily without squinting.

`if ( its a group or its a trusted contact, or its from us )` then its good 👌

Reply:
tbh I have no idea what that check actually check. This whole check is existing code and I figure I'd reuse the same